Content.Shared/Construction/Steps/MaterialConstructionGraphStep.cs

```diff
@@ -25,12 +25,12 @@ public override void DoExamine(ExaminedEvent examinedEvent)
 
         public override bool EntityValid(EntityUid uid, IEntityManager entityManager)
         {
-            return entityManager.TryGetComponent(uid, out SharedStackComponent? stack) && stack.StackTypeId == MaterialPrototypeId && stack.Count >= Amount;
+            return entityManager.TryGetComponent(uid, out SharedStackComponent? stack) && stack.StackTypeId == MaterialPrototypeId;
         }
 
         public bool EntityValid(EntityUid entity, [NotNullWhen(true)] out SharedStackComponent? stack)
         {
-            if (IoCManager.Resolve<IEntityManager>().TryGetComponent(entity, out SharedStackComponent? otherStack) && otherStack.StackTypeId == MaterialPrototypeId && otherStack.Count >= Amount)
+            if (IoCManager.Resolve<IEntityManager>().TryGetComponent(entity, out SharedStackComponent? otherStack) && otherStack.StackTypeId == MaterialPrototypeId)
```

Comment:
I completely missed that regular construction works out of separate code from initial (probably should have inferred that from "Initial" in the file name, d'oh). Do you think it's worthwhile to update the validation there too, to support the changes for considering all stacks you have?

From the looks of it, the "validate everything and then consume everything" model only makes sense for initial, since normal steps by nature must be done one-by-one.